pkg/pkg.bzl

```diff
@@ -31,15 +33,32 @@ def _quote(filename, protect = "="):
     """Quote the filename, by escaping = by \\= and \\ by \\\\"""
     return filename.replace("\\", "\\\\").replace(protect, "\\" + protect)
 
+
 def _pkg_tar_impl(ctx):
     """Implementation of the pkg_tar rule."""
 
     # Files needed by rule implementation at runtime
     files = []
 
+    if ctx.attr.package_file_name:
+        output_name = ctx.attr.package_file_name
+        package_variables = ctx.attr.package_variables[PackageVariablesInfo]
+        if package_variables:
+            output_name = output_name.format(**package_variables.values)
+        elif ctx.attr.package_file_name.find('{') >= 0:
+            fail('attribute package_file_name requires package_variables')
+        output_file = ctx.actions.declare_file(output_name)
+        ctx.actions.symlink(
+            output = ctx.outputs.out,
+            target_file = output_file
+        )
```

Comment:
Good catch. Now both are in outputs if needed.

The interaction is that
- if we do not use package_file_name.
  - only one file should be created.
- if we do use package_file_name
   - write only one instance of the archive, symlink the other
   - users want to pick up the file by either name.

I choose to make the explicit file the one with the potentially unpredictable name.
One could read the symlink for the predicable name and find the other out.
This works for the test case I have made, but I might not be seeing the case where the special
name should be the symlink and the real file should be the out.
It may not matter. If people want precision, they should be using the `PackageArtifactInfo` provider to pick the correct name.

Reply:
Gotcha, thanks!